Framework/AODMerger/src/aodStrainer.cxx

```diff
@@ -0,0 +1,207 @@
+    return -1;
+  }
+
+  TList* keyList = inputFile->GetListOfKeys();
+  keyList->Sort();
+
+  for (auto key1 : *keyList) {
+    if (((TObjString*)key1)->GetString().EqualTo("metaData")) {
+      auto metaDataCurrentFile = (TMap*)inputFile->Get("metaData");
+      outputFile->cd();
+      metaDataCurrentFile->Write("metaData", TObject::kSingleKey);
+    }
+
+    if (((TObjString*)key1)->GetString().EqualTo("parentFiles")) {
+      auto parentFilesCurrentFile = (TMap*)inputFile->Get("parentFiles");
+      outputFile->cd();
+      parentFilesCurrentFile->Write("parentFiles", TObject::kSingleKey);
+    }
+
+    if (!((TObjString*)key1)->GetString().BeginsWith("DF_") || dis(gen) > downsampling) {
+      continue;
+    }
+
+    auto dfName = ((TObjString*)key1)->GetString().Data();
+    if (verbosity > 0) {
+      printf("  Processing folder %s\n", dfName);
+    }
+    outputDir = outputFile->mkdir(dfName);
+    auto folder = (TDirectoryFile*)inputFile->Get(dfName);
+    auto treeList = folder->GetListOfKeys();
+
+    treeList->Sort();
+
+    // purging keys from duplicates
+    for (auto i = 0; i < treeList->GetEntries(); ++i) {
+      TKey* ki = (TKey*)treeList->At(i);
+      for (int j = i + 1; j < treeList->GetEntries(); ++j) {
+        TKey* kj = (TKey*)treeList->At(j);
+        if (std::strcmp(ki->GetName(), kj->GetName()) == 0 && std::strcmp(ki->GetTitle(), kj->GetTitle()) == 0) {
+          if (ki->GetCycle() < kj->GetCycle()) {
+            printf("    *** FATAL *** we had ordered the keys, first cycle should be higher, please check");
+            exitCode = 5;
+          } else {
+            // key is a duplicate, let's remove it
+            treeList->Remove(kj);
+            j--;
+          }
+        } else {
+          // we changed key, since they are sorted, we won't have the same anymore
+          break;
+        }
+      }
+    }
+
+    std::list<std::string> foundTrees;
+
+    for (auto key2 : *treeList) {
+      auto treeName = ((TObjString*)key2)->GetString().Data();
+      bool found = (std::find(foundTrees.begin(), foundTrees.end(), treeName) != foundTrees.end());
+      if (found == true) {
+        printf("    ***WARNING*** Tree %s was already merged (even if we purged duplicated trees before, so this should not happen), skipping\n", treeName);
+        continue;
+      }
+      bool foundTable = false;
+      for (auto const& table : listOfTables) {
+        if (table == removeVersionSuffix(treeName)) {
+          foundTrees.push_back(treeName);
+          foundTable = true;
+          break;
+        }
+      }
+      if (!foundTable) {
+        if (verbosity > 2) {
+          printf("    Skipping tree %s\n", treeName);
+        }
+        continue;
+      }
+
+      auto inputTree = (TTree*)inputFile->Get(Form("%s/%s", dfName, treeName));
+      bool fastCopy = (inputTree->GetTotBytes() > 10000000); // Only do this for large enough trees to avoid that baskets are too small
```

Comment:
Here we can always do fast copy. As we anyway, do not merge it with another tree in this step. So you can always put fast.